handbook: add Go test naming convention (#980)
sourcegraph/about: handbook/engineering/testing.md

- [Distribution team](distribution/index.md) owns build and test infrastructure.
- [Web team](web/index.md) owns any tests that are driven through the browser.

## Conventions

> - **Naming tests in Go code.** We strive to for the following conventions when naming Go test functions:

Author: ```suggestion
- **Naming tests in Go code.** We strive to follow these conventions when naming Go test functions:
```

>   - General rule: avoid using `_` immediately after `Test`. Instead, if testing a function like `fooBar`, the test is called TestFooBar and not `Test_fooBar`. 

Reviewer: ```suggestion
  - General rule: avoid using `_` immediately after `Test`. Instead, if testing a function like `fooBar`, the test is called `TestFooBar` and not `Test_fooBar`. 
```
